Remotion/Data/DomainObjects.UnitTests/Persistence/Rdbms/Parameters/ObjectIDDataParameterDefinitionFactoryTest.cs

```diff
@@ -0,0 +1,145 @@
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using Moq;
+using NUnit.Framework;
+using Remotion.Data.DomainObjects.Persistence;
+using Remotion.Data.DomainObjects.Persistence.Configuration;
+using Remotion.Data.DomainObjects.Persistence.Rdbms.Model;
+using Remotion.Data.DomainObjects.Persistence.Rdbms.Model.Building;
+using Remotion.Data.DomainObjects.Persistence.Rdbms.Parameters;
+using Remotion.Data.DomainObjects.Queries;
+using Remotion.Data.DomainObjects.UnitTests.Factories;
+using Remotion.Data.DomainObjects.UnitTests.Persistence.Configuration;
+using Remotion.Development.UnitTesting;
+
+namespace Remotion.Data.DomainObjects.UnitTests.Persistence.Rdbms.Parameters;
+
+[TestFixture]
+public class ObjectIDDataParameterDefinitionFactoryTest : StandardMappingTest
+{
+  private Mock<IDataParameterDefinitionFactory> _nextDataParameterDefinitionFactoryMock;
+
+  [SetUp]
+  public override void SetUp ()
+  {
+    base.SetUp();
+
+    _nextDataParameterDefinitionFactoryMock = new Mock<IDataParameterDefinitionFactory>();
+  }
+
+  [Test]
+  public void Initialize ()
+  {
+    var storageProviderDefinition = new TestableStorageProviderDefinition("name", Mock.Of<IStorageObjectFactory>());
+    var storageTypeInformationProvider = Mock.Of<IStorageTypeInformationProvider>();
+    var storageSettings = Mock.Of<IStorageSettings>();
+
+    var result = new ObjectIDDataParameterDefinitionFactory(storageProviderDefinition, storageTypeInformationProvider, storageSettings, _nextDataParameterDefinitionFactoryMock.Object);
+
+    Assert.That(result.StorageProviderDefinition, Is.SameAs(storageProviderDefinition));
+    Assert.That(result.StorageTypeInformationProvider, Is.SameAs(storageTypeInformationProvider));
+    Assert.That(result.StorageSettings, Is.SameAs(storageSettings));
+  }
+
+  [Test]
+  public void CreateDataParameterDefinition_WithObjectIDInSameProviderDefinition ()
+  {
+    var storageProviderDefinition = DomainObjectIDs.Customer1.StorageProviderDefinition;
+    var otherObjectID = DomainObjectIDs.Customer2;
+
+    var storageTypeInformationStub = new Mock<IStorageTypeInformation>();
+
+    var storageTypeInformationProviderStub = new Mock<IStorageTypeInformationProvider>();
+    storageTypeInformationProviderStub
+        .Setup(_ => _.GetStorageTypeForID(true))
+        .Returns(storageTypeInformationStub.Object);
+
+    var factory = new ObjectIDDataParameterDefinitionFactory(
+        storageProviderDefinition,
+        storageTypeInformationProviderStub.Object,
+        StorageSettings,
+        _nextDataParameterDefinitionFactoryMock.Object);
+
+    _nextDataParameterDefinitionFactoryMock
+        .Setup(mock => mock.CreateDataParameterDefinition(It.IsAny<QueryParameter>()))
```

Comment:
stylistic mix and mingle of `mock` and `_` naming is something I try to avoid at least within the same method.